pkg/upgrade/upgrader/jiva_upgrade.go

```diff
@@ -470,38 +444,52 @@ func (j *jivaVolumeOptions) replicaUpgrade(openebsNamespace string) error {
 
 	statusObj.Phase = utask.StepErrored
 
-	// Scaling down controller deployment before patching replica deployment
-	// if the replica is not upgraded already.
-	if j.replicaObj.version == currentVersion {
-		err = scaleTargetDeploy(j.controllerObj.name, j.ns, 0)
+	if currentVersion < "1.9.0" {
+		err = j.migrateReplica(openebsNamespace)
 		if err != nil {
-			statusObj.Message = "failed to scale down target depoyment"
+			statusObj.Message = "failed to migrate replica to openebs namespace"
 			statusObj.Reason = strings.Replace(err.Error(), ":", "", -1)
 			j.utaskObj, uerr = updateUpgradeDetailedStatus(j.utaskObj, statusObj, openebsNamespace)
 			if uerr != nil && isENVPresent {
 				return uerr
 			}
-			return errors.Wrap(err, "failed to scale down target depoyment")
+			return errors.Wrap(err, "failed to migrate replica to openebs namespace")
 		}
-	}
+	} else {
```

Comment:
this code structure doesn't look neat..
better to allow old code to get executed by reducing code in new path as provided in comments there
Same applies for target also

Reply:
its also required to allow executing existing code path as 'patch' may keep changing as we make new releases..
By combining upgrade code into migration path, we will miss the patches that need to be done in new releases.